lib/utils/test_utils.go

```diff
@@ -32,25 +34,17 @@ func NewTestDir(t *testing.T) string {
 
 // NewTestBasePath create new test data directory
 func NewTestBasePath(t *testing.T, name string) string {
-	testDir := path.Join(TestDir, t.Name())
-	basePath := path.Join(testDir, name)
-
-	err := os.Mkdir(TestDir, os.ModePerm)
-	if err != nil && !PathExists(TestDir) {
-		fmt.Println(fmt.Errorf("failed to create test directory: %s", err))
-	}
+	t.Helper()
 
-	err = os.Mkdir(testDir, os.ModePerm)
-	if err != nil && !PathExists(testDir) {
-		fmt.Println(fmt.Errorf("failed to create test directory: %s", err))
-	}
+	basepathTmp := path.Join(t.TempDir(), t.Name(), name)
+	// create the subdirectories all the subdirectories
+	// os.MkdirAll is equivalent to mkdir -p /dir/subdir1/subdir2
```

Comment:
I don't think that's necessary, one can always check out `os.MkdirAll` documentation if they want.

```suggestion
```